packages/flutter/test/material/dropdown_test.dart

```diff
@@ -3052,17 +3052,180 @@ void main() {
+              autofocus: true,
+              value: value,
+              items: options.map((String s) => DropdownMenuItem<String>(
+                value: s,
+                disabled: s == 'disabled',
+                child: Text(s),
+              )).toList(),
+              onChanged: (String? v) => setState(() => value = v),
+              focusNode: focusNode,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Pump a frame for autofocus to take effect.
+    await tester.pump();
+    expect(focusNode.hasPrimaryFocus, isTrue);
+
+    // Open dropdown.
+    await tester.tap(find.text('first').hitTestable());
+    await tester.pumpAndSettle();
+
+    // The first element should have the focus
+    expect(Focus.of(tester.element(find.text('first').hitTestable())).hasPrimaryFocus, isTrue);
+
+    // Tap on a disabled item.
+    await tester.tap(find.text('disabled').hitTestable());
+    await tester.pump();
+
+    // The first element of the dropdown list should retain the focus
+    expect(Focus.of(tester.element(find.text('first').hitTestable())).hasPrimaryFocus, isTrue);
+  });
+
+  testWidgets('disabled item should not be focusable when navigated using arrow keys', (WidgetTester tester) async {
+    final FocusNode focusNode = FocusNode(debugLabel: 'DropdownButton');
+    final List<String> options = <String>['first', 'disabled', 'second'];
+    String? value = options.first;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) => DropdownButton<String>(
+              autofocus: true,
+              value: value,
+              items: options.map((String s) => DropdownMenuItem<String>(
+                value: s,
+                disabled: s == 'disabled',
+                child: Text(s),
+              )).toList(),
+              onChanged: (String? v) => setState(() => value = v),
+              focusNode: focusNode,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Pump a frame for autofocus to take effect.
+    await tester.pump();
+    expect(focusNode.hasPrimaryFocus, isTrue);
+
+    // Open dropdown.
+    await tester.tap(find.text('first').hitTestable());
+    await tester.pumpAndSettle();
+
+    // Moving focus one down.
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+    await tester.pump();
+    // second should have the focus.
+    expect(Focus.of(tester.element(find.text('second').hitTestable())).hasPrimaryFocus, isTrue);
+
+    // Moving focus one up.
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+    await tester.pump();
+    // now first should have the focus.
+    expect(Focus.of(tester.element(find.text('first').hitTestable())).hasPrimaryFocus, isTrue);
+  });
+
+  testWidgets('an enabled and a disabled item can have the same value', (WidgetTester tester) async {
```

Comment:
Maybe change this to check to see if disabled items are correctly ignored by the menu system (for example, if there are 10 disabled items with "value" defined). 

On the other hand, maybe we do not want to allow disabled items to have values in the first place, since it does not provide it with any useful meaning. Also, this can easily be a place for developer to write bugs in their code (ie. they dynamically create their menu items, and somehow they write incorrect logic for value generation and deciding if the item is disabled/enabled. This would give them quite the headache to debug).

Reply:
I have added assertions so that [value] and [onTap] need to be `null` if [disabled] is `true`